core/station/impl/src/factories/requests/change_canister.rs

```diff
@@ -116,24 +117,81 @@ impl Execute for ChangeCanisterRequestExecute<'_, '_> {
                     self.request.operation.clone(),
                 ))
             }
+        }
+    }
+}
 
-            ChangeCanisterTarget::UpgradeCanister(canister_id) => {
-                CHANGE_CANISTER_SERVICE
-                    .install_canister(
-                        canister_id,
-                        CanisterInstallMode::Upgrade(None),
-                        &self.operation.input.module,
-                        self.operation.input.arg.clone(),
-                    )
-                    .await
-                    .map_err(|err| RequestExecuteError::Failed {
-                        reason: format!("failed to upgrade canister {}: {}", canister_id, err),
-                    })?;
+pub struct ChangeManagedCanisterRequestCreate;
 
-                Ok(RequestExecuteStage::Completed(
-                    self.request.operation.clone(),
-                ))
-            }
-        }
+impl Create<ChangeManagedCanisterOperationInput> for ChangeManagedCanisterRequestCreate {
+    fn create(
+        request_id: UUID,
+        requested_by_user: UUID,
+        input: CreateRequestInput,
+        operation_input: ChangeManagedCanisterOperationInput,
+    ) -> Result<Request, RequestError> {
+        let request = Request::new(
+            request_id,
+            requested_by_user,
+            Request::default_expiration_dt_ns(),
+            RequestOperation::ChangeManagedCanister(ChangeManagedCanisterOperation {
+                arg_checksum: operation_input.arg.as_ref().map(|arg| {
+                    let mut hasher = Sha256::new();
+                    hasher.update(arg);
+                    hasher.finalize().to_vec()
+                }),
+                module_checksum: {
+                    let mut hasher = Sha256::new();
+                    hasher.update(&operation_input.module);
+                    hasher.finalize().to_vec()
+                },
+                input: operation_input.into(),
+            }),
+            input
+                .execution_plan
+                .map(Into::into)
+                .unwrap_or(RequestExecutionPlan::Immediate),
+            input
+                .title
+                .unwrap_or_else(|| "ChangeManagedCanister".to_string()),
+            input.summary,
+        );
+
+        Ok(request)
+    }
+}
+
+pub struct ChangeManagedCanisterRequestExecute<'p, 'o> {
```

Comment:
ideally this would use dependency injection for the `CHANGE_CANISTER_SERVICE` here instead of the inline call

Reply:
i meant something like [this](https://github.com/dfinity/orbit/blob/fa4771d49c1f453325d00c5c968af18fbe930a69/core/station/impl/src/factories/requests/edit_request_policy.rs#L58)

Reply:
I see. Fixed for both cases here: https://github.com/dfinity/orbit/pull/270/commits/79d308c089e1a4a3717a7805b9d1d7fb7ec7909b